pandas-stubs/core/dtypes/missing.pyi

```diff
@@ -22,7 +24,7 @@ def isna(obj: Series) -> Series[bool]: ...
 @overload
 def isna(obj: Index | list | ArrayLike) -> np.ndarray: ...
```

Comment:
Could change the return type of this overload to `npt.NDArray[np.bool_]`

Reply:
Without rerunning stubgen on a fresh clone of Pandas repository, was there a way for me to see this improvement ? Is there a way to develop the intuition ?

Reply:
> Without rerunning stubgen on a fresh clone of Pandas repository, was there a way for me to see this improvement ? Is there a way to develop the intuition ?

Don't think even `stubgen` would figure that out.  We just learn as we go.  In this case, we know that when the results is an `np.ndarray` that contains only `boolean`, we can use a more specific type, which we always try to do whenever we can.

Reply:
Writing type annotations is a mix of reading the documentation, looking at the implementation, and using automated tools like stubgen/pyright. In this case, we also have those annotations in the implementation (the `NaTType` is missing there too): https://github.com/pandas-dev/pandas/blob/ce143a2fb0a41e78c85d5311d0e159552d651b00/pandas/core/dtypes/missing.py#L99